Makefile.toml

```diff
@@ -0,0 +1,64 @@
+[env]
+CARGO_MAKE_EXTEND_WORKSPACE_MAKEFILE = true
+
+[tasks.format]
+install_crate = "rustfmt"
+command = "cargo"
+args = ["fmt", "--", "--emit=files"]
+
+[tasks.clean]
+command = "cargo"
+args = ["clean"]
+
+[tasks.build]
+command = "cargo"
+args = ["build"]
+
+[tasks.test]
+workspace = false
+command = "cargo"
+args = ["nextest", "run", "--workspace", "${@}"]
+
+[tasks.pre-clone]
+workspace = false
+description = "Clean up existing directory"
+script = '''
+if [ -d "./hdp-cairo" ]; then
+    echo "Removing existing hdp-cairo directory..."
+    rm -rf ./hdp-cairo
+else
+    echo "No existing directory to remove."
+fi
+'''
+
+[tasks.clone]
+workspace = false
+description = "Clone the repository"
+dependencies = ["pre-clone"]
```

Comment:
instead of having pre-clone u can just ignore error on the cloning with `ignore_errors = true`